### PR TITLE
Feature 8162 add reason for deletion to confirmation dialogue

### LINKED ISSUE #8162: Add reason for deletion to confirmation dialogue [3]
### Feature Description

For GDPR compliance reasons it is necessary to document why data was deleted by a user.
The confirmation popup when deleting needs to be extended to allow this.
(This applies for deleting a single entity as well as using the bulk edit mode to delete data)

### Proposed Change

Rework the popup that shows when a user is trying to delete data and add a possibility to select a reason or enter your own.

The dropdown contains the vollowing values:

|Index|de_DE|en_US|
|:-------------:|-------------|-------------|
|0|Löschen auf Anforderung der betroffenen Person nach DSGVO|Deletion request by affected person according to GDPR|
|1|Löschen auf Anforderung einer anderen Behörde|Deletion request by another authority|
|2|Entität ohne Rechtsgrund angelegt|Entity created without legal reason|
|3|Abgabe des Vorgangs wegen Nicht-Zuständigkeit|Responsibility transferred to another authority|
|4|Löschen von Doubletten|Deletion of duplicate entries|
|5|Anderer Grund|Other reason|

If the user selects "Other Reason" in the dropdown menu a new text field appears below the dropdown where a user can enter their own reason. Entering text in the field is mandatory to be able to confirm the deletion of the data.
The field counts as filled in when the length of the trimmed string is greater than 0.

If the user has not selected a value in the dropdown and tries to confirm by clicking the "Yes" button the following error message appears:
| Locale|Value|
| ------------- |-------------|
|de_DE|Bitte wählen Sie einen Grund aus.|
|en_US|Please select a reason.|

If the user has selected "Other Reason" but did not enter anything in the text field and tries to confirm by clicking the "Yes" button the following error message appears:
| Locale|Value|
| ------------- |-------------|
|de_DE|Bitte geben Sie einen Grund an.|
|en_US|Please specify the reason.|

Other localization:

| Locale|Value|
| ------------- |-------------|
|de_DE|Löschen bestätigen|
|en_US|Confirm deletion|

| Locale|Value|
| ------------- |-------------|
|de_DE|Sie sind im Begriff diese Daten löschen zu lassen.  Dieser Vorgang kann nicht rückgängig gemacht werden.|
|en_US|You are about to mark this data for deletion. This action is irreversible.|

| Locale|Value|
| ------------- |-------------|
|de_DE|Grund|
|en_US|Reason|

| Locale|Value|
| ------------- |-------------|
|de_DE|Anderer Grund|
|en_US|Other reason|

| Locale|Value|
| ------------- |-------------|
|de_DE|Möchten Sie die Daten wirklich löschen?|
|en_US|Are you sure you want to delete this data?|

| Locale|Value|
| ------------- |-------------|
|de_DE|Nein|
|en_US|No|

| Locale|Value|
| ------------- |-------------|
|de_DE|Ja|
|en_US|Yes|

- [x] Add the two new fields (enum and text) to the DeletableAdo

Additional things to do here:

- [x]  In the UI of a deleted entity, we need to display when the manually deleted entity will be automatically permanently deleted. For other deletion reference types this has already been implemented in https://github.com/hzi-braunschweig/SORMAS-Project/issues/7240
- [x]  In the UI of a deleted entity, we need to display the reason why it has been deleted (at bottom of the form)

### Additional Information

Mockup:
2022-02-28_15h14_22

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #8162

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java [hunk 1]
```diff
@@ -588,6 +588,7 @@ public class CaseDataDto extends SormasToSormasShareableDto {
 	private Map<String, String> externalData;
 	private boolean deleted;
 	private DeletionReason deletionReason;
+	@Size(max = FieldConstraints.CHARACTER_LIMIT_TEXT, message = Validations.textTooLong)
```
Comment (BarnaBartha) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java:591]: this needs to be added to all other entities

**Review by BarnaBartha — CHANGES_REQUESTED** [review]
looks good except the comment + at least one test should also assert the deletion reason has been persisted somehove

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseDataDto.java:591]: changed

**Comment by sergiupacurariu** [comment]
tests added

**Review by BarnaBartha — APPROVED** [review]

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Change the details text in the ReST resources to "Deleted via ReST call"

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**